test/test_jit_cuda_fuser.py

```diff
@@ -4185,6 +4185,47 @@ def t(x_kj, idx_kj, sbf):
         t_jit = torch.jit.script(t)
         self._run_helper(t_jit, t, lookup_tv, indies_tv, sbf)
 
+    @unittest.skipIf(not RUN_NVFUSER, "requires CUDA")
+    @unittest.skipIf(GRAPH_EXECUTOR != ProfilingMode.PROFILING,
+                     "Requires fusion optimization pass to be effective")
+    def test_index_select_function(self):
+        def t(x: torch.Tensor, y: torch.Tensor, ind: torch.Tensor):
+            o = torch.mul(x, y)
+            o = torch.index_select(o, 0, ind)
+            return o
+
+        x = torch.randn([68, 128], dtype=torch.float, device="cuda").requires_grad_()
+        y = torch.randn_like(x).requires_grad_()
+        ind = torch.randint(0, 68, (130,), device="cuda").to(dtype=torch.int)
+        grad = torch.randn([130, 128], dtype=torch.float, device="cuda")
+
+        ref_x = x.detach().requires_grad_()
+        ref_y = y.detach().requires_grad_()
+        o = t(ref_x, ref_y, ind)
+        o.backward(grad)
+
+        t_jit = torch.jit.script(t)
+        jit_o = t_jit(x, y, ind)
+        jit_o.backward(grad)
+        jit_o = t_jit(x, y, ind)
+        jit_o.backward(grad)
+        jit_o = t_jit(x, y, ind)
+        jit_o.backward(grad)
+        x.grad.zero_()
+        y.grad.zero_()
+        jit_o = t_jit(x, y, ind)
+        jit_o.backward(grad)
+        self.assertEqual(ref_x.grad, x.grad)
+        self.assertEqual(ref_y.grad, y.grad)
+        self.assertTrue(self._compare("comparing output failed", o, jit_o, 1e-5))
+        self.assertGraphContainsExactly(t_jit.graph_for(x, y, ind), FUSION_GUARD, 1, consider_subgraphs=True)
```

Comment:
nitpick: can we switch this from using `_run_trainign_helper` instead?

in that function, we are not checking grad, let's add that as well. :bow:

Reply:
I updated `_run_training_helper` function to include grad checking. As it is called by `test_gelu`, and `test_dropout_training`, I have tested them all to make sure I didn't break anything.